multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/OIDCInfoProvider.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.keycloak.server;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Map;
+import org.eclipse.che.commons.proxy.ProxyAuthenticator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OIDCInfoProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OIDCInfoProvider.class);
+
+  private String tokenEndPoint;
+  private String userInfoEndpoint;
+  private String JWKS_URI;
+  private String endSessionEndpoint;
+
+  public void requestInfo(String wellKnownEndpoint) {
```

Comment:
what about having a static builder method that can be reused from KeycloakSettings constructor?